### ISSUE TITLE
Add a no-doc linter 

### ISSUE THREAD

**Issue by lread** [clj-kondo/clj-kondo#1197]
**Is your feature request related to a problem? Please describe.**
I'd like be warned when I am using a 3rd party library's internal undocumented var(s).
I'd also like to be able to deliberately quelch this warning on a case by case basis.

This feature request understands that only some libraries mark which vars represent their documented public API.

**Technical background**
Codox introduced the concept of tagging `ns`es and vars with `^:no-doc` metadata to indicate they are not part of a documented public API.  
When a `ns` is tagged with `^:no-doc` metadata, all vars within the namespace are implicitly tagged with `^:no-doc`.
Codox also considers `^:skip-wiki`, for legacy reasons, to be equivalent to `^:no-doc`.

As far as I understand it, clj-kondo already collects `:no-doc` metadata.

**Linter name**
I think `:no-doc` would be a good familiar name, and clearly suggests to those in the `^:no-doc` know, what it is all about.

If a solution also incorporates #455, maybe something more generic like `:internal-api`.

**Linter use cases**
Here are some use cases, I'll use rewrite-clj as an example library.

**case 1 - I want to warned if I reference any no-doc var outside rewrite-clj**
This would exclude, for rewrite-clj, reporting of any no-doc calls to rewrite-clj.

```Clojure
{:linters {:no-doc {:exclude ["rewrite-clj\\.*"]}}}
```
**case 2 - I want to be warned if I call any no-doc outside my lib with exceptions**
But what if I wanted to call say I single no-doc function from rewrite-clj to a 3rd party lib?

```Clojure
{:linter {:no-doc {:exclude ["rewrite-clj\\.*"
                             a.third.party.ns/a-no-doc-fn]}}}
```

Let's add another 3rd party ns, and another 3rd party regex:
```Clojure
{:linter {:no-doc {:exclude ["rewrite-clj\\.*"
                             a.third.party.ns/a-no-doc-fn
                             another.third.party.ns
                             "and\\.more\\.*"]}}}
```

**Linter message**
How about this? What we found and what it means:
`#'a.thirdparty.lib.ns.impl/calc is tagged with no-doc, meaning that it is not considered part of a public API`

**Describe alternatives you've considered**
I've not, ping me if you want me to think more.

**Additional context**
I'm not sure I understand the implications of all the mechanics, but from a naive perspective the above seems like it would work.

**Comment by lread** [comment]
Here's a detail to be aware of:

Codox allows you to affect what gets documented (i.e. what defines your public API) via [codox options](https://github.com/weavejester/codox#source-files).
For example, by default, vars that look like record constructors are excluded, which gives them an implicit `:no-doc`.

Cljdoc-analyzer does not currently give any such control when used in the context of cljdoc.  
But it does also exclude vars that look like record constructors.

**Comment by borkdude** [comment]
I think we should include more conventions like `:skip-wiki` and `.impl.` in the namespace. The name should probably be something like `:internal-api-call` or so?

**Comment by borkdude** [comment]
Yes
